cmd/clusterctl/cmd/delete.go

```diff
@@ -17,15 +17,71 @@ limitations under the License.
 package cmd
 
 import (
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
+type deleteOptions struct {
+	kubeconfig           string
+	forceDeleteNamespace bool
+	forceDeleteCRD       bool
+	deleteAll            bool
+}
+
+var dd = &deleteOptions{}
+
 var deleteCmd = &cobra.Command{
-	Use:   "delete",
-	Short: "Delete a cluster API resource",
-	Long:  `Delete a cluster API resource with one command`,
+	Use:   "delete [providers]",
+	Short: "Delete one or more provider from the management cluster",
+	Long: LongDesc(`
+		Deletes one or more providers from the management cluster.`),
+
+	Example: Examples(`
+		# Deletes the AWS provider and the kubeadm provider
+		# Please note that this imply the cancellation of everything related to the provider except the 
```

Comment:
Hopefully, now it is more clear

This deletes all the provider components that were originally deployed (including Deployment, Secrets/ConfigMaps, RBAC rules etc), except CRDs and Namespaces.

The rationale behind this is to avoid the deletion of any user-created object unless explicitly requested to do so.